qiskit_optimization/algorithms/qrao/quantum_random_access_optimizer.py

```diff
@@ -0,0 +1,312 @@
+
+        Raises:
+            ValueError: If the maximum number of variables per qubit is not 1, 2, or 3.
+            TypeError: If the provided minimum eigensolver does not support auxiliary operators.
+        """
+        if max_vars_per_qubit not in (1, 2, 3):
+            raise ValueError("max_vars_per_qubit must be 1, 2, or 3, but was {max_vars_per_qubit}.")
+        self._max_vars_per_qubit = max_vars_per_qubit
+        self.min_eigen_solver = min_eigen_solver
+        # Use ``QuadraticProgramToQubo`` to convert the problem to a QUBO.
+        if rounding_scheme is None:
+            rounding_scheme = SemideterministicRounding()
+        self._rounding_scheme = rounding_scheme
+        self._converters = QuadraticProgramToQubo(
+            penalty=penalty,
+        )
+
+    @property
+    def min_eigen_solver(self) -> MinimumEigensolver:
+        """Return the minimum eigensolver."""
+        return self._min_eigen_solver
+
+    @min_eigen_solver.setter
+    def min_eigen_solver(self, min_eigen_solver: MinimumEigensolver) -> None:
+        """Set the minimum eigensolver."""
+        if not min_eigen_solver.supports_aux_operators():
+            raise TypeError(
+                f"The provided MinimumEigensolver ({type(min_eigen_solver)}) "
+                "does not support auxiliary operators."
+            )
+        self._min_eigen_solver = min_eigen_solver
+
+    @property
+    def max_vars_per_qubit(self) -> int:
+        """Return the maximum number of variables per qubit."""
+        return self._max_vars_per_qubit
+
+    @max_vars_per_qubit.setter
+    def max_vars_per_qubit(self, max_vars_per_qubit: int) -> None:
+        """Set the maximum number of variables per qubit."""
+        self._max_vars_per_qubit = max_vars_per_qubit
+
+    @property
+    def rounding_scheme(self) -> RoundingScheme:
+        """Return the rounding scheme."""
+        return self._rounding_scheme
+
+    @rounding_scheme.setter
+    def rounding_scheme(self, rounding_scheme: RoundingScheme) -> None:
+        """Set the rounding scheme."""
+        self._rounding_scheme = rounding_scheme
+
+    def get_compatibility_msg(self, problem: QuadraticProgram) -> str:
+        """Checks whether a given problem can be solved with this optimizer.
+
+        Checks whether the given problem is compatible, i.e., whether the problem can be converted
+        to a QUBO, and otherwise, returns a message explaining the incompatibility.
+
+        Args:
+            problem: The optimization problem to check compatibility.
+
+        Returns:
+            A message describing the incompatibility.
+        """
+        return QuadraticProgramToQubo.get_compatibility_msg(problem)
+
+    def solve_relaxed(
+        self,
+        encoding: QuantumRandomAccessEncoding,
+    ) -> tuple[MinimumEigensolverResult, RoundingContext]:
+        """Solve the relaxed Hamiltonian given by the encoding.
```

Comment:
In this case the `max_vars_per_qubit ` as given in the above constructor I guess is irrelevant since the problem would be encoded however the encoding was setup. Should we note this here? Really the only thing its using from the instance is the minimum eigensolver - if that were passed here this could be a class/static method - more just noting this.